instrumentation/opentelemetry/github.com/gin-gonic/hypergin/gin_test.go

```diff
@@ -0,0 +1,86 @@
+package hypergin
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/hypertrace/goagent/instrumentation/opentelemetry/internal"
+	sdkhttp "github.com/hypertrace/goagent/sdk/instrumentation/net/http"
+	"go.opentelemetry.io/otel/trace"
+	"gotest.tools/assert"
+)
+
+// inspired in https://github.com/jcchavezs/httptest-php/blob/e6a65c73/src/HttpTest/HttpTestServer.php#L150
+func findAvailablePort() (int, error) {
```

Comment:
While I truly appreciate you use this piece of code, you can use the package `httptest-php` is inspired on which is https://pkg.go.dev/net/http/httptest

Reply:
Ah okay - I copied this almost entirely from the mux sample but subbed in gin where appropriate - I'll update